### DISCUSSION TITLE
APPUI Viewport Frontstage   Tools

### DISCUSSION THREAD

**Discussion by paprophet** [iTwin/itwinjs-core#201] [Q&A] [answered]
APPUI Viewport Frontstage Sample have some  tools ,for example, section view 、 identify element 、snapmode、view change tools(top、right) on the right top of page 
1.i want use this tools to Viewer Features page(for example Display Styles),
but i do not know : what class code and how use  this tools
2.if i use APPUI Viewport Frontstage Sample main page , how control those tools (old tools positions、remove、add new tools and i can control new tools position )

**Comment by NancyMcCallB** [comment, marked as answer]
The AppUi Viewport Frontstage Sample uses the standard Tool and Navigation widgets (named BasicToolWidget and Basic Navigation Widget, respectively). If you'd like to add tools, you have two options:
1) Add them to the standard widgets at load time, as we illustrate in the Add Toolbar Button sample
or
2) Create your own widget(s) and put your custom widgets in the frontstage definition. You can copy the source from imodeljs/ui/src/ui-framework/widgets/BasicToolWidget.tsx, change the name (to avoid confusion) and change the toolbar items that are returned from the getHorizontalToolbarItems() and getVerticalToolbarItems() methods to whatever you want. You'll find a list of tool defintions in imodeljs/core/frontend/src/public/locales/english/CoreTools.json.

If you want to remove or rearrange tools in the widgets, you'll have to go with option 2 and create your own widgets.

This seems like a reasonable sample to create, so I'll try to get one in place in the next week or so.

Let me know if you run into trouble!
